FESS-GUI/mainwindow.cpp

```diff
@@ -360,10 +374,54 @@ void MainWindow::on_goButton_clicked()  //when you hit the go button
     ui->textBrowser->append(QString("Flywheel controlled to %1 RPM,"
                                     " %2 rad/sec<sup>2</sup>, %3 rad/sec<sup>3</sup>"
                                     " at %4")
-                            .arg(expectedVelocity).arg(expectedAcceleration).arg(expectedJerk)
+                            .arg(targetVelocity).arg(targetAcceleration).arg(currentExpectedJerk)
                             .arg(QTime::currentTime().toString()));
 }
 
+/*this function manages the slope. It is called every 10ms when you click the go button,
+  and stops when you get to your target velocity */
+void MainWindow::velocitySlope()
+{
+    double intervalIncrement = 1000 / velocitySlopeTimer->interval(); //get the correct increment
+
+    if(currentExpectedVelocity <= targetVelocity){ //if the target velocity is greater than the current
+        currentExpectedVelocity += currentExpectedAcceleration / intervalIncrement; //increment the velocity
+        if(currentExpectedVelocity >= targetVelocity){
+            velocitySlopeTimer->stop();
+            currentExpectedVelocity = targetVelocity; //in case the numbers don't round nicely
+            currentExpectedAcceleration = 0;
```

Comment:
I think you should stop the accelerationSlopeTimer as well if you're setting expectedAcceleration to 0. If you don't, I think that could lead to acceleration being set to 0, but it hadn't reached its target yet, so it gets set to 0 and starts climbing again until it reaches the target, and it never gets reset to 0